app/logic/gossip_node_logic.rb

```diff
@@ -40,6 +41,21 @@ def update_nodes
     end
   end
 
+  def set_inactive_nodes_status
+    lambda do |p|
+      active_nodes_identities = p.payload[:current_nodes]&.map { |cn| cn["identityPubkey"] }
+      GossipNode.where(network: p.payload[:network]).each do |gn|
```

Comment:
perhaps it Could work without running through each gn? I. e. You could do `GossipNode.where(network: p.payload[:network]).where.not(account: active_nodes_identities).update_all(is_active: false)`